src/DataBuilder.php

```diff
@@ -754,13 +754,18 @@ protected function scrub($arr, $fields, $replacement = '*')
             return null;
         }
 
-        $scrubber = function (&$val, $key) use ($fields, $replacement, $arr) {
-
+        $scrubber = function (&$val, $key) use ($fields, $replacement, &$scrubber) {
             if (in_array($key, $fields)) {
                 $val = str_repeat($replacement, 8);
             }
+
+            if (is_array($val)) {
+                array_walk($val, $scrubber);
+            }
         };
-        array_walk_recursive($arr, $scrubber);
```

Comment:
Oh, I just saw @MichaelMackus comment above. I think that making the code to error out when passing other types than array is a good idea, since they are not going to get scrubbed anyway. My reasoning is that if I pass some data to be scrubbed (e.g. an object) and the library doesn't support this type, it's a pretty critical issue. We don't want the users to be passing objects and thinking they got scrubbed when they didn't.